tests/integration/security/testdata/requestauthn/remotehttps.yaml.tmpl

```diff
@@ -0,0 +1,15 @@
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: request-authn
+spec:
+  selector:
+    matchLabels:
+      app: {{ .dst }}
+  jwtRules:
+  - issuer: "test-issuer-1@istio.io"
+    jwksUri: "https://jwt-server.{{ .Namespace }}.svc.cluster.local:8443/jwks"
```

Comment:
on the cert I see `DNS:*.localhost.com, DNS:*.local, DNS:localhost`. So how does this work?

Reply:
request made to *.local or *.localhost.com or localhost can be validated with this cert.

Reply:
`*.local` is not supposed to match `x.y.local` IIUC. It should only matches up to one label?

Reply:
'*' would signify any number of char, so I guess it will allow multiple dots.

Reply:
good catch, the `*.local` won't match with the `jwt-server.<namespace.svc.cluster.local` in certificate, see https://en.wikipedia.org/wiki/Wildcard_certificate

I think the easiest fix is to generate the certificate with DNS name `jwt-server` and `localhost` and modify to deploy it in the same namespace as istiod, which should be istio-system by default, and change the RequestAuthentication jwksUri to just `https://jwt-server:8443`, you could omit the namespace part as long as they are in the same namespace.

Reply:
I think you can do `*.*.svc.cluster.local` too. Probably not a great idea though... Also would be ideal to not encode `cluster.local` if possible because we eventually want to make it tested with that customized. But that is probably a long way off, do not need to solve it here.

Reply:
I thought only the left most `*` will work as the wildcard matching? I did a quick test by modifying the unit test and it seems so.

Reply:
@yangminzhu I tried to do it by adding jwt-server in the dns, but it is failing. And one more thing, in the test we are doing the deployment as well right ? https://github.com/istio/istio/blob/master/tests/integration/security/jwt_test.go#:~:text=jwtserver%20%3A%3D%20applyyaml(%22..%2F..%2F..%2Fsamples%2Fjwt-server%2Fjwt-server.yaml%22%2C%20ns)

Reply:
could you clarify what is failing? you need to change that line to pass the `istio-system` ns.

Reply:
this is the error which I'm receiving - `failed calling a (cluster=cluster-0)->'http://b.test-ns1-1-37638.svc.cluster.local:8095//valid-token-forward-remote-jwks': call failed from a (cluster=cluster-0) to http://b.test-ns1-1-37638.svc.cluster.local:8095/valid-token-forward-remote-jwks (using http)`

Reply:
@howardjohn correct me if I'm wrong, I just read about wildcards and found out multiple wildcard doesn't work - https://en.wikipedia.org/wiki/Wildcard_certificate#:~:text=A%20cert%20with,*.*.domain.com